src/ci/pgo.sh

```diff
@@ -128,13 +154,20 @@ python3 $CHECKOUT/x.py build --target=$PGO_HOST --host=$PGO_HOST \
 
 # Here we're profiling the `rustc` frontend, so we also include `Check`.
 # The benchmark set includes various stress tests that put the frontend under pressure.
-# The profile data is written into a single filepath that is being repeatedly merged when each
-# rustc invocation ends. Empirically, this can result in some profiling data being lost.
-# That's why we override the profile path to include the PID. This will produce many more profiling
-# files, but the resulting profile will produce a slightly faster rustc binary.
-LLVM_PROFILE_FILE=${RUSTC_PROFILE_DIRECTORY_ROOT}/default_%m_%p.profraw gather_profiles \
-    "Check,Debug,Opt" "All" \
-    "externs,ctfe-stress-5,cargo-0.60.0,token-stream-stress,match-stress,tuple-stress,diesel-1.4.8,bitmaps-3.1.0"
+if isLinux; then
+    # The profile data is written into a single filepath that is being repeatedly merged when each
+    # rustc invocation ends. Empirically, this can result in some profiling data being lost. That's
+    # why we override the profile path to include the PID. This will produce many more profiling
+    # files, but the resulting profile will produce a slightly faster rustc binary.
+    LLVM_PROFILE_FILE=${RUSTC_PROFILE_DIRECTORY_ROOT}/default_%m_%p.profraw gather_profiles \
+        "Check,Debug,Opt" "All" \
+        "externs,ctfe-stress-5,cargo-0.60.0,token-stream-stress,match-stress,tuple-stress,diesel-1.4.8,bitmaps-3.1.0"
+else
+    # On windows, we don't do that, and use the default profraw merging behavior.
```

Comment:
Can we make a note here why this is? (disk space limitations?)

Reply:
Exactly, disk space limitations: I tried it and it filled the disk completely, [as can be seen here](https://github.com/rust-lang/rust/pull/96978#issuecomment-1157325789).

I'll add a comment to mention this.

Reply:
Done.